### ISSUE TITLE
vmauth additional per user metrics for SLO/SLI calculation. User metric labels.

### ISSUE THREAD

**Issue by f41gh7** [VictoriaMetrics/VictoriaMetrics#5565]
### Is your feature request related to a problem? Please describe

 Currently, vmauth exposes:
`vmauth_user_requests_total`
`vmauth_user_request_duration_seconds`
`vmauth_user_concurrent_requests_limit_reached_total`

 It's not enough to calculate service level. 

 Also, those metrics have only `username` as label. It's hard to aggregate given metrics and requires to encode additional into name label. 

### Describe the solution you'd like

 It'd be great to expose request errors metric, when request to all backends failed (retry doesn't count as error for this case). 

 Proposed name - `vmauth_user_requests_errors_total`

It allows to calculate error budget for services behind vmauth.

 In addition, it'd be great to have configurable labels for per user metrics.

```yaml
users:
- username: tenant1
  password: "***"
  name: dev-1
 # vmauth_user_requests_total{username="dev-1", users_group="dev",users_backend_location="dc-0"}
 # vmauth_user_requests_errors_total{username="dev-1", users_group="dev",users_backend_location="dc-0"}
  metricLabels:
  - users_group="dev"
  - users_backend_location="dc-0"
  url_map:
  - src_paths:
    - "/api/v1/write"
    url_prefix: "http://vminsert-backend-dc-0:8480/insert/1/prometheus/"
- username: accounting-access-1
  password: "***"
  name: accountant-1
  metricLabels:
  - users_group="accounting"
  - users_backend_location="dc-15"
  url_map:
  - src_paths:
    - "/api/v1/write"
    url_prefix: "http://vminsert-dc-1:8480/insert/1/prometheus/"
- username: accounting-access-2
  password: "***"
  name: accountant-2
  metricLabels:
  - users_group="accounting"
  - users_backend_location="dc-15"
  url_map:
  - src_paths:
    - "/api/v1/write"
    url_prefix: "http://vminsert-dc-1:8480/insert/1/prometheus/"
```

**Comment by valyala** [comment]
The commit b3598ba2c1f9ae5dc6cd8fbc122315c1572bb95d adds `vmauth_user_request_backend_errors_total` per-user metric and `vmauth_unauthorized_user_request_backend_errors_total` metric for unauthorized users. This commit also adds an ability to specify additional labels for per-user metrics via `metric_labels` section at user config level. See [these docs](https://docs.victoriametrics.com/vmauth.html#monitoring) for more details.

This commit will be included in the next release. In the mean time it is possible to build [`vmauth`](https://docs.victoriametrics.com/vmauth.html) from this commit according to [these docs](https://docs.victoriametrics.com/vmauth.html#how-to-build-from-sources) and verify whether it works as intended.

**Comment by valyala** [comment]
FYI, `vmauth` supports metrics listed [here](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5565#issuecomment-1902485555) starting from [v1.97.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the feature request as done.

**Comment by valyala** [comment]
See also the [follow-up pull request](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6078), which stops tracking concurrency limit errors at `vmauth_user_request_backend_errors_total` metric, since these errors are already tracked at `vmauth_user_concurrent_requests_limit_reached_total` metric. This simplifies monitoring **real errors** at backend by the owner of `vmauth` service. This pull request will be included in the next release.

**Comment by hagen1778** [comment]
This change was included in [v1.101.0](https://docs.victoriametrics.com/changelog/#v11010) release.
